### PR TITLE
Temporary access provision for AppLens

### PR DESCRIPTION
**Pull request by SharmaAjay19**
Temporary Access Provision for AppLens. It defaults to 7 days and can be availed one time only.
Screenshots:
When user does not have access to applens
image

When temporary access is approved
image

And user is redirected to home page
image

When user has already used one-time temporary access
image

### CODE REVIEW

File: ApplensBackend/Controllers/TemporaryAccessController.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                if (userId != null)
+                {
+                    var result = await _cosmosDBHandler.GetItemAsync(userId);
+                    if (result == null)
+                    {
+                        TemporaryAccessUser newUser = new TemporaryAccessUser()
+                        {
+                            Id = userId,
+                            AccessStartDate = DateTime.UtcNow,
+                            UserPrincipalName = userId
```
Comment (rekhaswaminathan) [inline ApplensBackend/Controllers/TemporaryAccessController.cs]: `Id` and `UserPrincipalName` are taking same value `userId`. Do we need both?

File: ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        private string Key;
+        private string DatabaseId;
+        private string CollectionId;
+        private string PartitionKey = "/UserPrincipalName";
+        private DocumentClient client;
+
+        public CosmosDBHandler(IConfiguration configuration)
+        {
+            Endpoint = configuration["ApplensTemporaryAccess:Endpoint"];
+            Key = configuration["ApplensTemporaryAccess:Key"];
```
Comment (rekhaswaminathan) [inline ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs]: This `Key` is actually meant for connecting to CosmosDB right? Should we rename to it something like "AppLensCosmosDB". Same for `ApplensTemporaryAccess:Endpoint`

File: ApplensBackend/Controllers/TemporaryAccessController.cs (hunk 1 above)
Comment (SharmaAjay19) [reply ApplensBackend/Controllers/TemporaryAccessController.cs]: Yes, we need a partition key as well as unique id in cosmos db.

File: ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs (hunk 2 above)
Comment (SharmaAjay19) [reply ApplensBackend/Services/CosmosDBHandler/CosmosDBHandler.cs]: Yes we will be using this terminology for the detectors cosmos. Here the sole purpose of the handler is for temporary access hence the name.

**Pull request merged**